src/Microsoft.ML.Api/DataViewConstructionUtils.cs

```diff
@@ -14,7 +14,7 @@ namespace Microsoft.ML.Runtime.Api
     /// <summary>
     /// A helper class to create data views based on the user-provided types.
     /// </summary>
-    internal static class DataViewConstructionUtils
+    public static class DataViewConstructionUtils
     {
         public static IDataView CreateFromList<TRow>(IHostEnvironment env, IList<TRow> data,
```

Comment:
>IList [](start = 75, length = 5)

Since we in this file, @tfinley@gmail.com  do you have any opinion about IList?
I would personally convert it to List.

Reply:
Is there a reason why you would prefer `List` here?

So the reason I would prefer a specific implementer of an interface rather than the interface itself is to avoid having to go through the virtual function table. (Which is why it's often better to write `Foo<TBar>(TBar biz) where TBar : IBar` vs. `Foo(IBar biz)`.) In this specific case though, I don't think it's worth it. You're ultimately going to be dealing with an `IEnumerator<TRow>` *anyway* for most operations on this structure, and there it's unfortunately unavoidable, so I care a bit less. But perhaps there's something to this I'm not appreciating.

---
In reply to: [187205023](https://github.com/dotnet/machinelearning/pull/106#discussion_r187205023) [](ancestors = 187205023)